src/api/java/mekanism/api/chemical/Chemical.java

```diff
@@ -226,44 +378,52 @@ public boolean is(TagKey<Chemical> tag) {
      */
     @Deprecated(forRemoval = true, since = "10.7.9")
     public Stream<TagKey<Chemical>> getTags() {
-        return getAsHolder().tags();
+        return builtInRegistryHolder.tags();
     }
 
     /**
-     * Helper method to get the holder for this chemical. Unlike {@link net.minecraft.world.item.Item#builtInRegistryHolder()} and similar, this looks up the holder from
-     * the registry when called.
+     * Helper method to get the holder for this chemical.
      *
      * @since 10.6.0
+     * @deprecated If a holder is necessary use {@link #builtInRegistryHolder()}
      */
+    @Deprecated(forRemoval = true, since = "10.7.11")
     public Holder<Chemical> getAsHolder() {
-        return MekanismAPI.CHEMICAL_REGISTRY.wrapAsHolder(this);
+        return builtInRegistryHolder();
+    }
+
+    /**
+     * Intrusive holder, similar to vanilla this is deprecated and will eventually be moved away from.
+     *
+     * @since 10.7.11
+     */
+    @Deprecated
```

Comment:
don't deprecate it, mark it as internal api

Reply:
The reason I had deprecated it was to follow how vanilla marks all their builtInRegistryHolders, as they do want to remove them. I can definitely mark it as internal, but I wonder if we should mark it as both, as ideally we also want to avoid using it even internally if we can say reference the holder through the chemical stack.

Reply:
and how exactly is vanilla going to remove it? The fact is, it's getting used anyway so in this iteration it's not deprecated as there is no other way to do it (or if there is, you should do that)

Reply:
I have gone ahead and removed the builtinRegistryHolder entirely from this PR as after all the refactoring I had done, it ended up actually being pretty easy to remove.